<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProgramPageDelphi</class>
 <widget class="QWidget" name="ProgramPageDelphi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>630</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="9" column="1" colspan="4">
      <widget class="QCheckBox" name="check_show_advanced_options">
       <property name="text">
        <string>Show Advanced Options</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1" colspan="4">
      <widget class="QComboBox" name="combo_start_mode">
       <item>
        <property name="text">
         <string>Executable</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Compile on RED Brick (fpc)</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Start Mode:</string>
       </property>
       <property name="buddy">
        <cstring>combo_start_mode</cstring>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="5">
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="5" column="1" colspan="4">
      <widget class="QComboBox" name="combo_main_source_file">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="insertPolicy">
        <enum>QComboBox::NoInsert</enum>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="label_working_directory">
       <property name="text">
        <string>Working Directory:</string>
       </property>
       <property name="buddy">
        <cstring>combo_working_directory</cstring>
       </property>
      </widget>
     </item>
     <item row="11" column="0">
      <widget class="QLabel" name="label_options">
       <property name="text">
        <string>FPC Options:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="buddy">
        <cstring>list_options</cstring>
       </property>
      </widget>
     </item>
     <item row="10" column="1" colspan="4">
      <widget class="QComboBox" name="combo_working_directory">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="insertPolicy">
        <enum>QComboBox::NoInsert</enum>
       </property>
       <item>
        <property name="text">
         <string>.</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="11" column="1" colspan="4">
      <widget class="ExpandingListWidget" name="list_options"/>
     </item>
     <item row="13" column="1">
      <widget class="QPushButton" name="button_add_option">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item row="13" column="2">
      <widget class="QPushButton" name="button_remove_option">
       <property name="text">
        <string>Remove</string>
       </property>
      </widget>
     </item>
     <item row="13" column="3">
      <widget class="QPushButton" name="button_up_option">
       <property name="text">
        <string>Up</string>
       </property>
      </widget>
     </item>
     <item row="13" column="4">
      <widget class="QPushButton" name="button_down_option">
       <property name="text">
        <string>Down</string>
       </property>
      </widget>
     </item>
     <item row="12" column="1" colspan="4">
      <widget class="QLabel" name="label_options_help">
       <property name="text">
        <string>This list of options will be passed to the Free Pascal Compiler (FPC).</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1" colspan="4">
      <widget class="QLabel" name="label_start_executable_help">
       <property name="text">
        <string>The executable has to be cross compiled for armhf.</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="1" colspan="4">
      <widget class="QLabel" name="label_start_compile_help">
       <property name="text">
        <string>In &quot;Compile Mode&quot; the program is compiled on the RED Brick. The Free Pascal Compiler will be called during the last step, after the upload has finished.</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="7" column="1" colspan="4">
      <widget class="QLabel" name="label_main_source_file_help">
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Specify the main source file. It will be given to the Free Pascal Compiler for compiling.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_main_source_file_type">
       <property name="text">
        <string>Main Source File Type:</string>
       </property>
       <property name="buddy">
        <cstring>combo_main_source_file_type</cstring>
       </property>
      </widget>
     </item>
     <item row="6" column="1" colspan="4">
      <widget class="QComboBox" name="combo_main_source_file_type">
       <property name="currentIndex">
        <number>1</number>
       </property>
       <item>
        <property name="text">
         <string>All Files</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Delphi/Lazarus Files (*.pas, *.pp)</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_executable">
       <property name="text">
        <string>Executable:</string>
       </property>
       <property name="buddy">
        <cstring>combo_executable</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1" colspan="4">
      <widget class="QComboBox" name="combo_executable">
       <property name="editable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="1" colspan="4">
      <widget class="QLabel" name="label_executable_help">
       <property name="text">
        <string>The specified executable will be executed.</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_main_source_file">
       <property name="text">
        <string>Main Source File:</string>
       </property>
       <property name="buddy">
        <cstring>combo_main_source_file</cstring>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ExpandingListWidget</class>
   <extends>QListWidget</extends>
   <header location="global">brickv.plugin_system.plugins.red.program_utils</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>combo_start_mode</tabstop>
  <tabstop>combo_executable</tabstop>
  <tabstop>combo_main_source_file</tabstop>
  <tabstop>combo_main_source_file_type</tabstop>
  <tabstop>check_show_advanced_options</tabstop>
  <tabstop>combo_working_directory</tabstop>
  <tabstop>list_options</tabstop>
  <tabstop>button_add_option</tabstop>
  <tabstop>button_remove_option</tabstop>
  <tabstop>button_up_option</tabstop>
  <tabstop>button_down_option</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
